chapters/interface.tex

```diff
@@ -585,45 +585,48 @@ \section{Function-Compatibility or Function-Subtyping for Functions}\label{funct
 
 \section{Type Compatible Expressions}\label{type-compatible-expressions}
 
-Certain expressions consist of an operator applied to two or more type compatible subexpressions (\lstinline!A! and \lstinline!B!), including binary operators, e.g. \lstinline!A + B!, \lstinline!if!-expressions, e.g.\ \lstinline!if x then A else B!, and array expressions, e.g.\ \lstinline!{A, B}!.
-The resulting type of the expression in case of two type compatible subexpressions \lstinline!A! and \lstinline!B! is defined as follows:
+Certain expressions consist of an operator applied to two or more subexpressions (\lstinline!A! and \lstinline!B!).
+
+This includes:
+\begin{itemize}
+\item \lstinline!if!-expressions, e.g.\ \lstinline!if x then A else B!.
+\item Array expressions, e.g.\ \lstinline!{A, B}!
+\item Binary operators if both operands are of simple types, e.g.\ \lstinline!A + B!.
+Binary operators for other types are only defined for operator records, \cref{overloaded-binary-operations}, and do not necessarily require that the operands are type compatible with each other.
+\end{itemize}
+If the subexpressions satisfy the following restrictions they are called type compatible expressions.
+Otherwise the expression is illegal.
+The type of the full expression (e.g.\ \lstinline!if x then A else B!) is also defined below.
```

Comment:
This is what made me open #3235.  Let's hope we can merge the present PR first, meaning that later fixing #3235 will take care of aligning this new text with the rest of the document.